plugins/usertiming.js

```diff
@@ -0,0 +1,110 @@
+/**
+\file usertiming.js
+Plugin to collect metrics from the W3C User Timing API.
+For more information about User Timing,
+see: http://www.w3.org/TR/user-timing/
+
+This plugin is dependent on the UserTimingCompression library
+see: https://github.com/nicjansma/usertiming-compression.js
+UserTimingCompression must be loaded before this plugin's init is called.
+*/
+
+(function() {
+
+	BOOMR = BOOMR || {};
+	BOOMR.plugins = BOOMR.plugins || {};
+
+	if (BOOMR.plugins.UserTiming) {
+		return;
+	}
+
+	/**
+	 * @returns String compressed user timing data that occurred since the last call
+	 */
+	function getUserTiming() {
+		var map, res, now = BOOMR.now();
+
+		map = UserTimingCompression.getCompressedUserTiming(impl.options);
+		res = UserTimingCompression.compressForUri(map);
+		impl.options["from"] = now;
+
+		return res;
+	}
+
+	var impl = {
+		complete: false,
+		initialized: false,
+		supported: false,
+		options: {"from": 0},
+
+		addEntriesToBeacon: function() {
+			var r;
+
+			if (this.complete) {
+				return;
+			}
+
+			BOOMR.removeVar("usertiming");
+			r = getUserTiming();
+			if (r) {
+				BOOMR.info("User Timing API entries found since the last beacon", "usertiming");
+				BOOMR.addVar({
+					"usertiming": r
+				});
+			}
+
+			this.complete = true;
+			BOOMR.sendBeacon();
+		},
+
+		clearMetrics: function(vars) {
+			if (vars.hasOwnProperty("usertiming")) {
+				BOOMR.removeVar("usertiming");
+			}
+			this.complete = false;
+		}
+	};
+
+	BOOMR.plugins.UserTiming = {
+		init: function(config) {
+			var p = BOOMR.window.performance;
+
+			BOOMR.utils.pluginConfig(impl, config, "UserTiming");
+
+			if (impl.initialized) {
+				return this;
+			}
+
+			// Check that we have getEntriesByType and that the required UserTimingCompression library is available
+			if (p && typeof p.getEntriesByType === "function" && typeof UserTimingCompression !== "undefined") {
```

Comment:
might have to use `BOOMR.window.UserTimingCompression`, since it's possible that the library was loaded into the host page, but that boomerang itself is running in an anonymous iframe.

I think it might also be good to add the `UserTimingCompression` test first so that we can log an error telling the developer what to do (ie, to load the library).